client/modules/IDE/pages/IDEView.jsx

```diff
@@ -48,6 +48,26 @@ function WarnIfUnsavedChanges() {
 
   const currentLocation = useLocation();
 
+  // beforeunload handles closing or refreshing the window.
+  useEffect(() => {
+    const handleUnload = (e) => {
+      // See: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#browser_compatibility
```

Comment:
Thanks for adding this, this is great! It would definitely be helpful to have more of these references/comments throughout the codebase!